server/src/core/server/services/users/emailPremodFilter.ts

```diff
@@ -104,6 +104,26 @@ const emailIsAnAliasOfExistingUser = async (
   return similarUsers.length > 0;
 };
 
+const emailIsDomainAlias = (email: string | undefined) => {
+  if (!email) {
+    return false;
+  }
+  const emailSplit = email.split("@");
+  if (emailSplit.length < 2) {
+    return false;
+  }
+
+  const domain = emailSplit[1].trim().toLowerCase();
+
+  let periodCount = 0;
+  for (const char of domain) {
+    if (char === ".") {
+      periodCount++;
+    }
+  }
+  return periodCount > 1;
+};
```

Comment:
I believe this will catch too many legitimate email addresses, including institutional emails (@mail.school.edu, @army.mil.us) and second-level domains like @gmail.co.uk. Can we do the same thing we do for email aliases, where we strip out the subdomain, then check if that domain is banned, then premod?